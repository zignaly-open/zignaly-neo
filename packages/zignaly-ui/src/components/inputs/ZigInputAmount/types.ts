import { FilledTextFieldProps } from "@mui/material/TextField/TextField";

export type Coin = {
  coin: string;
  balance: string | number;
};

export type ZigInputAmountProps = Omit<FilledTextFieldProps, "variant" | "error"> & {
  wide?: boolean;
  error?: boolean | string;
  disabled?: boolean;
  id?: string;
  coin?: string | Coin;
  tokenOptions?: any;
  onTokenChange?: (token: any) => void;
  showMaxButton?: boolean;
  /**
   * Extra info to show under the input.
   * By default it will show the balance, min and max, if those values are provided.
   * To hide them, set them to false.
   * To change the labels, set them to a string.
   *
   * To add additional items, use the `others` array property:
   * { value: "value", label: "label" } or JSX.Element
   */
  extraInfo?: InputExtraInfo;
  /**
   * Wrap info if more than specified number of items. Default is 2. Set to true/false to force.
   */
  wrapExtraInfo?: number | boolean;
  /**
   * Custom onMax callback but onChange will already be called with the max value.
   */
  onMax?: () => void;
  /**
   * Pressing the MAX button will trigger this value to be set.
   * Also showed as info under the input.
   * */
  balance?: string | number;
  /** Showed as info. */
  min?: string | number;
  /** Showed as info */
  max?: string | number;
  className?: string;
  labelInline?: boolean;
  withCoinSelector?: boolean;
};

export type InputExtraInfoItem = { value: string | number; label: string };
export type InputExtraInfoFalseableItem = string | InputExtraInfoItem | false;

export type InputExtraInfoProps = {
  balance?: string | number;
  min?: string | number;
  max?: string | number;
  extraInfo?: InputExtraInfoObject;
};

export type InputExtraInfoObject = {
  balance?: InputExtraInfoFalseableItem;
  min?: InputExtraInfoFalseableItem;
  max?: InputExtraInfoFalseableItem;
  others?: InputExtraInfoFalseableItem[];
  wrapExtraInfo?: number | boolean;
};

export type InputExtraInfo = JSX.Element | false | InputExtraInfoObject;
