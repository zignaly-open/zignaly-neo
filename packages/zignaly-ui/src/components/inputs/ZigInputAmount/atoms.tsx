import { Box, Divider } from "@mui/material";
import ZigTypography from "components/display/ZigTypography";
import React from "react";
import { InputExtraInfoFalseableItem, InputExtraInfoItem, InputExtraInfoProps } from "./types";
import { NumericFormat } from "react-number-format";
import BigNumber from "bignumber.js";

const DEFAULT_ITEMS = {
  balance: "Available:",
  min: "Min:",
};

// const DEFAULT_ITEMS = [{}];

const extractItem = (item: InputExtraInfoFalseableItem, key?: keyof typeof DEFAULT_ITEMS) => {
  if (!item) return null;

  if (typeof item === "object") {
    return {
      value: item.value,
      label: item.label ?? (key ? DEFAULT_ITEMS[key] : ""),
    };
  } else {
    return { value: item, label: key ? DEFAULT_ITEMS[key] : "" };
  }
};

export const InputExtraInfo = ({
  balance,
  min,
  others,
  coin,
}: InputExtraInfoProps & {
  coin: string;
}) => {
  let items = [
    extractItem(balance ?? false, "balance"),
    extractItem(min ?? false, "min"),
    // { value: balance, text: DEFAULT_ITEMS.balance },
    // { value: min, text: LABELS.min },
    // { value: balance, text },
  ] as (InputExtraInfoItem | JSX.Element | null)[];

  // let items = [];
  if (others) {
    items = items.concat(
      others.map((item) =>
        React.isValidElement(item) ? item : extractItem(item as InputExtraInfoFalseableItem),
      ),
    );
  }
  console.log(items);

  // return (
  //   <Divider orientation="vertical" sx={{ backgroundColor: "red" }} variant="middle">
  //     <div>
  //       <ZigTypography variant="body2">
  //         {/* {LABELS.balance} */}
  //         {item.label}
  //         &nbsp;
  //         <ZigTypography variant="body2" color="neutral100">
  //           <NumericFormat
  //             value={item.value}
  //             displayType="text"
  //             thousandSeparator={true}
  //             suffix={` ${coin}`}
  //           />
  //         </ZigTypography>
  //       </ZigTypography>
  //     </div>
  //   </Divider>
  // );

  return (
    <Box
      display="flex"
      flexWrap="wrap"
      flexDirection={items.length <= 2 ? "row" : "column"}
      gap="6px"
    >
      {items.filter(Boolean).map((item, i) => (
        <React.Fragment key={i}>
          {React.isValidElement(item) ? (
            item
          ) : (
            <ZigTypography variant="body2">
              {/* {LABELS.balance} */}
              {(item as InputExtraInfoItem).label}
              &nbsp;
              <ZigTypography variant="body2" color="neutral100">
                <NumericFormat
                  value={(item as InputExtraInfoItem).value}
                  displayType="text"
                  thousandSeparator={true}
                  suffix={` ${coin}`}
                />
              </ZigTypography>
            </ZigTypography>
          )}
          {i === 0 && items.length === 2 && (
            <Divider
              orientation="vertical"
              sx={{ backgroundColor: "#35334a", m: "1px 22px" }}
              flexItem
            />
          )}
        </React.Fragment>
      ))}
    </Box>
  );
};
