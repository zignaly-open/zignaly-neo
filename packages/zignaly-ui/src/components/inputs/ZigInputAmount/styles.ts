import { Divider } from "@mui/material";
import { styled, Box } from "@mui/system";
import ZigButton from "../ZigButton";

export const Layout = styled(Box)<{
  withCoinSelector?: boolean;
  error: boolean;
  labelInline: boolean;
}>`
  border: ${({ withCoinSelector, theme }) =>
    withCoinSelector ? "none;" : `1px dotted ${theme.palette.neutral600};`};
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: ${(withCoinSelector) => (withCoinSelector ? "5px 24px 12px;" : "24px 24px 12px;")};
  position: relative;
  border-radius: 5px;

  ${({ error, theme, withCoinSelector }) =>
    error && !withCoinSelector && `border-color: ${theme.palette.redGraphOrError};`}
  ${({ labelInline }) =>
    labelInline &&
    `
      border-top: none;
      margin-top: 5px;
    `}

  && {
    .MuiInput-root {
      padding: 2px 24px;
      background: #171b30;
      ${({ withCoinSelector }) =>
        withCoinSelector &&
        `
        border-radius: 5px 0 0 5px;
        background: unset;
      `}

      border: ${({ withCoinSelector, theme }) =>
        withCoinSelector ? `1px dotted ${theme.palette.neutral600};` : `none;`};
      border-right: none;
    }

    .MuiInput-input {
      font-size: 30px;
      font-weight: 500;
      border-color: ${({ theme }) => theme.palette.neutral400};
      background: ${(withCoinSelector) => (withCoinSelector ? "unset;" : "#171b30;")};
      -webkit-text-fill-color: #878dd1 !important;
    }
  }
`;

export const DividerWrapper = styled(Box)<{ error?: boolean }>`
  padding: 8px 0;
  border-top: 1px dotted;
  border-bottom: 1px dotted;
  border-color: ${({ theme, error }) =>
    error ? theme.palette.redGraphOrError : theme.palette.neutral600};
  width: 100%;
  height: 100%;
`;

export const TopDivider = styled(Divider, {
  shouldForwardProp: (prop) => prop !== "error",
})<{ error: boolean }>`
  position: absolute;
  text-align: center;
  left: 0;
  right: 0;
  top: 0;
  transform: translateY(-50%);
  &:before {
    left: 1.5px;
  }
  &:after {
    right: 1.5px;
  }

  &:before,
  &:after {
    border-color: ${({ theme, error }) =>
      error ? theme.palette.redGraphOrError : theme.palette.neutral600};
    border-top-style: dotted;
  }
`;

export const MaxButton = styled(ZigButton)`
  padding: 3px 10px;
  border-radius: 13px;
  border: solid 1px #35334a;
  min-width: 44px;
  min-height: 30px;
  font-size: 12px;
  background: transparent;
  color: ${({ theme }) => theme.palette.neutral200};
`;
