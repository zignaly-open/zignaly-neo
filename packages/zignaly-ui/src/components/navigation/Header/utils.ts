// TODO: Move this for config or theme folder.
export const headerConfig = {
  height: "52px",
};
