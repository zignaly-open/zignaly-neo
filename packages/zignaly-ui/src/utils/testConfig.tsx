import { render } from "@testing-library/react";
import { ThemeProvider as ThemeProviderMui } from "@mui/material";
import { ThemeProvider, ThemeProviderMui as ThemeInheritorMui } from "../index";
import React from "react";
import { getZignalyThemeExport } from "../theme";
import mediaQuery from "css-mediaquery";
import { JestI18nextDecorator } from "./dummyI18nextProvider";

export const renderWithProvidersUi = (ui: JSX.Element, renderOptions = {}) => {
  const { mui, legacyStyledComponentsDoNotUse } = getZignalyThemeExport();
  return render(ui, {
    wrapper: ({ children }: { children: JSX.Element }) => (
      <JestI18nextDecorator>
        <ThemeProvider theme={legacyStyledComponentsDoNotUse}>
          <ThemeInheritorMui theme={mui}>
            <ThemeProviderMui theme={mui}>{children}</ThemeProviderMui>
          </ThemeInheritorMui>
        </ThemeProvider>
      </JestI18nextDecorator>
    ),
    ...renderOptions,
  });
};

const createMatchMedia = (width: number) => {
  return (query: string) => {
    return {
      matches: mediaQuery.match(query, { width }),
      media: "",
      addListener: () => {},
      removeListener: () => {},
      onchange: () => {},
      addEventListener: () => {},
      removeEventListener: () => {},
      dispatchEvent: () => true,
    };
  };
};

export const resizeScreenSize = (width: number) => {
  window.matchMedia = createMatchMedia(width);
};
