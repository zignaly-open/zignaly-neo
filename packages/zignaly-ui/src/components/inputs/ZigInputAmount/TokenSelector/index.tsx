import * as React from "react";
import { ReactElement, useState, useRef, useMemo } from "react";
import { useClickAway } from "react-use";
import { Layout, Container, Value, ArrowContainer, ValueContainer } from "./styles";
import { ReactComponent as CaretDownIcon } from "assets/icons/caret-down-icon.svg";
import { TokenSelectorProps } from "./types";
import { Coin } from "../types";
import { ZigCoinIcon } from "../../../../index";
import ZigSelect from "../../ZigSelect";

function TokenSelector({
  value,
  tokens = [],
  onSelectToken = () => null,
}: TokenSelectorProps<any>): ReactElement {
  const selectorRef = useRef(null);
  const [isActiveMenu, setMenuActive] = useState(false);
  const coinVal = typeof value === "object" ? value.coin : value ?? "";

  useClickAway(selectorRef, () => {
    setMenuActive(false);
  });

  /**
   * @var tokensAvailable
   * @description Stored variable showing the available tokens (filters the selected token to avoid duplication).
   */
  const tokensAvailable = useMemo(
    () => tokens.filter((token) => token.value !== coinVal),
    [tokens, value],
  );
  console.log(tokensAvailable);
  return (
    <Layout isActiveMenu={isActiveMenu} ref={selectorRef}>
      <Container onClick={() => setMenuActive(!isActiveMenu)}>
        {value && (
          <ValueContainer>
            {coinVal && <ZigCoinIcon coin={coinVal} size={"small"} />}
            <Value>{coinVal.toUpperCase()}</Value>
          </ValueContainer>
        )}
        <ArrowContainer>
          <CaretDownIcon width={"20px"} height={"10px"} />
        </ArrowContainer>
      </Container>
      <ZigSelect
        onChange={(token: Coin) => {
          onSelectToken(token);
        }}
        options={tokensAvailable}
      />
      {/*<Menu>*/}
      {/*  {tokensAvailable.length &&*/}
      {/*    tokensAvailable.map((token: Coin, index) => (*/}
      {/*      <Item*/}
      {/*        key={`--${index.toString()}`}*/}
      {/*        onClick={() => {*/}
      {/*          onSelectToken(token);*/}
      {/*          setMenuActive(false);*/}
      {/*        }}*/}
      {/*      >*/}
      {/*        {token.coin && <ZigCoinIcon coin={token.coin} size={"small"} />}*/}
      {/*        <Value>{token.coin.toUpperCase()}</Value>*/}
      {/*      </Item>*/}
      {/*    ))}*/}
      {/*</Menu>*/}
    </Layout>
  );
}

export default TokenSelector;
