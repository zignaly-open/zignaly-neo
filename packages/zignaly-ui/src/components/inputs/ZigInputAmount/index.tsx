import React, { forwardRef } from "react";
import { InputExtraInfoObject, ZigInputAmountProps } from "./types";
import { InputAdornment, Box } from "@mui/material";
import { ErrorMessage } from "../../display/ZigAlertMessage";
import ZigInput from "../ZigInput";
import { Layout, MaxButton, TopDivider } from "./styles";
import ZigTypography from "components/display/ZigTypography";
import { InputExtraInfo } from "./atoms";
import ZigCoinIcon from "../../display/ZigCoinIcon";
import { changeEvent } from "utils/event";
import BigNumber from "bignumber.js";
export { InputExtraInfo } from "./atoms";

const ZigInputAmount = forwardRef((props: ZigInputAmountProps, ref) => {
  const {
    label,
    error,
    wide,
    coin,
    extraInfo,
    onMax,
    placeholder = "0.0",
    wrapExtraInfo,
    balance,
    min,
    max,
    id,
    children,
    className = "",
    ...rest
  } = props;
  const coinVal = typeof coin === "object" ? coin.coin : coin ?? "";

  /**
   * Call custom onMax function or call onChange with max value event for RHF
   */
  const handleMax = () => {
    if (onMax) return onMax();
    if (props.onChange && balance !== "undefined") {
      props.onChange(
        changeEvent(
          props.name,
          max !== "undefined" && new BigNumber(max!).isLessThan(new BigNumber(balance!))
            ? max
            : balance,
        ) as React.ChangeEvent<HTMLInputElement>,
      );
    }
  };

  /**
   * Format the input value to keep only decimals.
   * We can't use type=number because it allows commas, which we replace by dots here.
   */
  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    let value = e.target.value;

    // Remove all non-numeric characters except the comma and dot
    value = value.replace(/[^0-9,\.]/g, "");

    // if the value ends with a comma, replace it with a dot
    if (value.endsWith(",")) {
      value = value.slice(0, -1) + ".";
    } else {
      // otherwise, remove all commas
      value = value.replace(/,/g, "");
    }

    // eslint-disable-next-line no-param-reassign
    e.target.value = value;
    props.onChange?.(e);
  };

  return (
    <Box display="flex" flexDirection="column" className={className}>
      <Layout display={wide ? "flex" : "inline-flex"} error={!!error}>
        <TopDivider error={!!error}>
          <ZigTypography color="neutral300" variant="body2">
            {label}
          </ZigTypography>
        </TopDivider>
        <Box display="flex" alignItems="center" gap={2} width={wide ? 1 : "auto"}>
          <Box display="flex" alignItems="center" gap={1}>
            <ZigCoinIcon size="small" coin={coinVal} />
            <ZigTypography color="neutral100" variant="h3">
              {coinVal}
            </ZigTypography>
          </Box>
          <ZigInput
            {...rest}
            id={id}
            inputRef={ref}
            type="string"
            wide={wide}
            label={null}
            sx={{ width: wide ? 1 : "auto" }}
            error={!!error}
            placeholder={placeholder}
            InputProps={{
              endAdornment:
                balance || handleMax ? (
                  <InputAdornment position="end">
                    <MaxButton variant="outlined" onClick={handleMax}>
                      Max
                    </MaxButton>
                  </InputAdornment>
                ) : null,
            }}
            onChange={handleChange}
          />
        </Box>
        <Box mt="16px" width={1}>
          <>
            {React.isValidElement(extraInfo) ? (
              extraInfo
            ) : extraInfo !== null ? (
              <InputExtraInfo
                coin={coinVal}
                min={min}
                max={max}
                balance={balance}
                extraInfo={extraInfo as InputExtraInfoObject}
              />
            ) : null}
            {children && (
              <Box display="flex" justifyContent="center" mt={3} mb={1}>
                {children}
              </Box>
            )}
          </>
        </Box>
      </Layout>
      {error && typeof error === "string" && (
        <Box alignSelf="flex-start" mt="11px">
          <ErrorMessage text={error} id={id && `${id}-error`} />
        </Box>
      )}
    </Box>
  );
});

export default ZigInputAmount;
