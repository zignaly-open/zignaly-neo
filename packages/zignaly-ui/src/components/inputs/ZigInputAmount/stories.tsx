import React, { useState } from "react";
import "@mui/system";
import { Meta, StoryObj } from "@storybook/react";
import ZigInputAmount from "./index";
import "@mui/system";
import ZigTypography from "components/display/ZigTypography";
import { ZigInputAmountProps } from "./types";

const meta = {
  title: "Inputs/ZigInputAmount",
  component: ZigInputAmount,
  argTypes: {
    label: {
      type: "string",
    },
    placeholder: {
      type: "string",
    },
    wide: {
      type: "boolean",
      defaultValue: false,
    },
    disabled: {
      type: "boolean",
      defaultValue: false,
    },
    defaultValue: {
      type: "string",
    },
    error: {
      type: "string",
      defaultValue: "",
    },
  },
} as Meta;
export default meta;

const ZigInputAmountWithState = (props: ZigInputAmountProps) => {
  // Sets the hooks for both the label and primary props
  const [value, setValue] = useState(props.value);
  // const [isPrimary, setIsPrimary] = useState(false);

  // Sets a click handler to change the label's value
  // const handleOnChange = () => {
  //   if (!isPrimary) {
  //     setIsPrimary(true);
  //     setValue("Primary");
  //   }
  // };
  return (
    <ZigInputAmount
      {...props}
      // onMax={
      //   typeof props.extraInfo === "object" && props.extraInfo !== null
      //     ? () => setValue(props.extraInfo?.balance)
      //     : null
      // }
      // onMax1={
      //   props.extraInfo && typeof props.extraInfo === "object" && "balance" in props.extraInfo
      //     ? () => setValue(props.extraInfo.balance)
      //     : undefined
      // }
      // onMax0={
      //   typeof props.extraInfo === "object" && "balance" in props.extraInfo
      //     ? () => setValue(props.extraInfo.balance)
      //     : null
      // }
      // onMax={
      //   props.extraInfo && props.extraInfo.balance ? () => setValue(props.extraInfo.balance) : null
      // }
      onMax={
        props.extraInfo && props.extraInfo?.balance
          ? () => setValue(props.extraInfo.balance)
          : undefined
      }
      value={value}
    />
  );
};

type Story = StoryObj<typeof meta>;

export const Default = { args: {} };
export const WithLabel: Story = {
  args: {
    label: "A label",
    coin: "USDT",
    extraInfo: {
      balance: 1000,
      min: { value: 100, label: "Min. deposit:" },
    },
  },
  render: (props) => <ZigInputAmountWithState {...props} />,
};

export const WithCustomExtraInfo = {
  args: {
    label: "A label",
    // coin: { coin: "USDT", balance: 10 },
    coin: "USDT",
    // extraInfo: (
    //   <InputExtraInfo balance={{ value: 1000, label: "Cuustom balance label" }} min={100} />
    // ),
    extraInfo: {
      balance: 1000,
      min: { value: 100, label: "Min. deposit:" },
      others: [
        {
          label: "Custom label:",
          value: 200,
        },
        <ZigTypography variant="body2" color="neutral100" key="1">
          Custom element
        </ZigTypography>,
      ],
    },
  },
};

// export const Default = Template.bind({});

// const Template: ComponentStory<typeof InputText> = (args) => <InputText {...args} />;

// export const TextInputer: ComponentMeta<typeof InputText> = Template.bind({});
// TextInputer.args = {
//   label: "Amount to Withdraw",
//   placeholder: "Amount to Withdraw",
// };

// export const Sensitive: ComponentMeta<typeof InputText> = Template.bind({});
// Sensitive.args = {
//   label: "Amount to Withdraw",
//   sensitive: true,
//   placeholder: "Amount to Withdraw",
// };

// export const TextareaInputer: ComponentMeta<typeof InputText> = Template.bind({});
// TextareaInputer.args = {
//   value:
//     "https://zignaly.com/api/signals.php?key=YOURSECRETKEY&type=entry&exchange=zignaly&pair=ethusdt&orderType=limit&positionSize=10&signalId=123&limitPrice=3420&takeProfitPercentage1=20&takeProfitAmountPercentage1=100&stopLossPercentage=-5",
//   multiline: true,
//   wide: true,
//   label: "Label",
// };

// export const Error: ComponentMeta<typeof InputText> = Template.bind({});
// Error.args = {
//   value: "https://zignaly com/api/",
//   wide: true,
//   error: "Invalid URL",
//   label: "Label",
// };

// export const LabelActionLink: ComponentMeta<typeof InputText> = Template.bind({});
// LabelActionLink.args = {
//   value: "",
//   wide: true,
//   labelAction: {
//     text: "Link to somewhere",
//     href: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
//   },
//   label: "Label",
// };

// export const LabelAction: ComponentMeta<typeof InputText> = Template.bind({});
// LabelAction.args = {
//   value: "",
//   wide: true,
//   labelAction: {
//     text: "Alert",
//     onClick: () => alert(),
//     tabIndex: -1,
//   },
//   label: "Label",
// };
