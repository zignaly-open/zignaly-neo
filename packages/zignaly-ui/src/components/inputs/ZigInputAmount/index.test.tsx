import React from "react";
import { renderWithProvidersUi } from "../../../utils/testConfig";
import { fireEvent, waitFor } from "@testing-library/react";
import ZigInputAmount from ".";

describe("components/inputs/ZigInputAmount", () => {
  it("should render without crashing", () => {
    const { container } = renderWithProvidersUi(<ZigInputAmount id={"test-id"} />);

    const input = container.querySelector("#test-id") as Element;
    expect(input).toBeVisible();
    const inputStyles = getComputedStyle(input);
    expect(input).toMatchSnapshot();
    expect(inputStyles).toMatchSnapshot();
  });

  it("should change value on input event", () => {
    const { container } = renderWithProvidersUi(<ZigInputAmount id={"test-id"} />);

    const input = container.querySelector("#test-id") as HTMLInputElement;
    fireEvent.change(input, { target: { value: "4" } });
    waitFor(() => {
      expect(input.value).toBe("4");
    });
  });

  it("should call handleMax when max button is clicked", () => {
    const handleMax = jest.fn();
    const { container } = renderWithProvidersUi(
      <ZigInputAmount id={"test-id"} onMax={handleMax} />,
    );

    const maxButton = container.querySelector("#test-id-max-button") as Element;
    fireEvent.click(maxButton);
    waitFor(() => {
      expect(handleMax).toHaveBeenCalled();
    });
  });

  it("should display error message when `error` prop is passed", () => {
    const { container } = renderWithProvidersUi(
      <ZigInputAmount id={"test-id"} error={"Test error"} />,
    );

    const errorMessage = container.querySelector("#test-id-error") as Element;

    expect(errorMessage).toBeInTheDocument();
    expect(errorMessage).toHaveTextContent("Test error");
  });
});
