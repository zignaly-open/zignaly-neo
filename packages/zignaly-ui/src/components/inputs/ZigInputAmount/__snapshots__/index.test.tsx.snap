// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/inputs/ZigInputAmount should render without crashing 1`] = `
<input
  aria-invalid="false"
  class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedEnd css-1fsdrue-MuiInputBase-input-MuiInput-input"
  data-testid="test-id"
  id="test-id"
  placeholder="0.0"
  type="string"
  value=""
/>
`;

exports[`components/inputs/ZigInputAmount should render without crashing 2`] = `
CSSStyleDeclaration {
  "0": "text-rendering",
  "1": "letter-spacing",
  "10": "color",
  "11": "outline",
  "12": "width",
  "13": "box-shadow",
  "14": "resize",
  "15": "border",
  "16": "box-sizing",
  "17": "background",
  "18": "height",
  "19": "margin",
  "2": "word-spacing",
  "20": "min-width",
  "21": "animation-name",
  "22": "animation-duration",
  "23": "visibility",
  "3": "text-transform",
  "4": "text-indent",
  "5": "text-shadow",
  "6": "display",
  "7": "text-align",
  "8": "padding",
  "9": "cursor",
  "_importants": {
    "animation-duration": "",
    "animation-name": "",
    "background": "",
    "border": "",
    "box-shadow": "important",
    "box-sizing": "",
    "color": "",
    "cursor": "",
    "display": "",
    "font": "",
    "height": "",
    "letter-spacing": "",
    "margin": "",
    "min-width": "",
    "outline": "",
    "padding": "",
    "resize": "",
    "text-align": "",
    "text-indent": "",
    "text-rendering": "",
    "text-shadow": "",
    "text-transform": "",
    "visibility": undefined,
    "width": "",
    "word-spacing": "",
  },
  "_length": 24,
  "_onChange": [Function],
  "_values": {
    "animation-duration": "10ms",
    "animation-name": "mui-auto-fill-cancel",
    "background": "none",
    "background-image": "none",
    "border": "0px",
    "border-bottom": "0px",
    "border-bottom-width": "0px",
    "border-left": "0px",
    "border-left-width": "0px",
    "border-right": "0px",
    "border-right-width": "0px",
    "border-top": "0px",
    "border-top-width": "0px",
    "border-width": "0px",
    "box-shadow": "none",
    "box-sizing": "content-box",
    "color": "currentColor",
    "cursor": "auto",
    "display": "block",
    "height": "1.4375em",
    "letter-spacing": "inherit",
    "margin": "0px",
    "margin-bottom": "0px",
    "margin-left": "0px",
    "margin-right": "0px",
    "margin-top": "0px",
    "min-width": "0",
    "outline": "none",
    "padding": "4px 0px 5px",
    "padding-bottom": "5px",
    "padding-left": "0px",
    "padding-right": "0px",
    "padding-top": "4px",
    "resize": "none",
    "text-align": "start",
    "text-indent": "0",
    "text-rendering": "auto",
    "text-shadow": "none",
    "text-transform": "none",
    "visibility": "visible",
    "width": "100%",
    "word-spacing": "normal",
  },
}
`;
